docassemble/ALToolbox/ThreePartsDate.py

```diff
@@ -538,7 +543,12 @@
   }}
 
   // TODO: Catch invalid alMax attr values for devs? Log in console? Make post MVP issue
-  let max_date = new Date(get_$original_date(field).attr('data-alMax').replace(/-/g, '/'));
+  let max_attr = get_$original_date(field).attr('data-alMax');
+  if (!max_attr) {{
+    // Validation should always succeed if no minimum given
+    return true;
+  }}
```

Comment:
I don't think that's true? The demo doesn't specify a max date for the example birth date at all (and it doesn't look like there's any available in the DOM.

```yaml
  - Birth date (today is valid): date_of_birth
    datatype: BirthDate
    required: True
    validation messages:
      required: This is a custom 'required' message.
```

If we do want to start enforcing the max attribute for birthdates, what is it supposed to do if there isn't a max attribute on the birthdate? I don't see any good examples of our error handling for developer errors in javascript.

Should it just keep the current PR's code, and have future dates be handled by the python validation, which always happens for Birth dates?

Reply:
It sounds to me like @plocket forgot that the birthdate validation was happening in Python, not JS. This seems good to merge to me.

Reply:
I believe the code in the DOM sets the max date for BirthDate to the current date as the default: https://github.com/SuffolkLITLab/docassemble-ALToolbox/blob/27858f6571361f47416982d04cc2e8c950fbe979/docassemble/ALToolbox/ThreePartsDate.py#L552-L554

Reply:
Gotcha, thanks plocket. I misunderstood what you were saying, and I got a little lost in the code and didn't see the line right below. I'll fix this in a new PR since this was merged too soon.